packages/svelte/src/internal/client/dev/hmr.js

```diff
@@ -20,6 +21,9 @@ export function hmr(source) {
 		block(() => {
 			const component = get(source);
 
+			// @ts-expect-error
+			check_target(new.target && component);
```

Comment:
this does the check even when the compiler option is set, which shouldn't happen in that case

Reply:
ah whoops, good catch. fixed. though it now fails with `props is undefined`, looking into that now

Reply:
kinda stumped by this tbh, any ideas?

Reply:
seems that checking for the truthiness of `new.target` (rather than doing an equality check) makes it work. i don't fully understand why the equality check was failing but i'll take it